File: app/src/main/java/com/example/sunkai/heritage/ConnectWebService/HandlePerson.java
package com.example.sunkai.heritage.ConnectWebService;

import android.support.annotation.Nullable;

import com.example.sunkai.heritage.Data.FocusData;
import com.example.sunkai.heritage.Activity.LoginActivity;
import com.example.sunkai.heritage.Data.OtherPersonData;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.ksoap2.serialization.SoapObject;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by sunkai on 2017-5-2.
 * 此类封装了个人中心页服务器请求的方法
 */


public class HandlePerson extends BaseSetting {

    @Nullable
    private static List<FocusData> Json_To_focusData(String json){
        try{
            JSONObject jsonObject=new JSONObject(json);
            JSONArray info=(JSONArray)jsonObject.get("Follow_Information");
            List<FocusData> datas=new ArrayList<>();
            for(int i=0;i<info.length();i++){
                FocusData data=new FocusData();
                JSONObject jsondata=(JSONObject)info.get(i);
                data.setFocusUserid(Integer.parseInt(jsondata.getString("focus_focusID")));
                data.setFocusFansID(Integer.parseInt(jsondata.getString("focus_fansID")));
                data.setName(jsondata.getString("USER_NAME"));
                datas.add(data);
            }
            return datas;
        }
        catch (JSONException e){
            e.printStackTrace();
        }
        return null;
    }
    @Nullable
    private static List<FocusData> Json_To_SearchData(String json){
        try{
            JSONObject jsonObject=new JSONObject(json);
            JSONArray searInfo=(JSONArray)jsonObject.get("searchInfo");
            List<FocusData> datas=new ArrayList<>();
            for(int i=0;i<searInfo.length();i++){
                JSONObject jsonData=(JSONObject)searInfo.get(i);
                FocusData data=new FocusData();
                data.setFolloweachother(false);
                data.setCheck(false);
                data.setName(jsonData.getString("user_name"));
                data.setFocusFansID(Integer.parseInt(jsonData.getString("id")));
                data.setFocusUserid(LoginActivity.Companion.getUserID());
                datas.add(data);
            }
            return datas;
        }
        catch (JSONException e){
            e.printStackTrace();
        }
        return null;
    }
    @Nullable
    private static OtherPersonData Json_To_OtherPersonData(int userID,String json){
        if(json==null)
            return null;
        try{
            JSONObject jsonObject=new JSONObject(json);
            String userName=jsonObject.getString("userName");
            int focusNumber=jsonObject.getInt("focusNumber");
            int fansNumber=jsonObject.getInt("fansNumber");
            int permisison=jsonObject.getInt("permission");
            return new OtherPersonData(userID,focusNumber,fansNumber,userName,permisison);
        }catch (JSONException e){
            e.printStackTrace();
        }
        return null;
    }
    public static boolean Update_User_Image(int userID,String image){
        methodName="Update_User_Image";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        soapObject.addProperty("image",image);
        String result=Get_Post(soapObject);
        return success.equals(result);
    }
    public static String Get_User_Image(int userID){
        methodName="Get_User_Image";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        if(error.equals(result))
            return null;
        return result;
    }
    public static int Get_Follow_Number(int userID){
        methodName="Get_Follow_Number";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        if(error.equals(result)||result==null){
            return 0;
        }
        return Integer.parseInt(result);
    }
    public static int Get_Fans_Number(int userID){
        methodName="Get_Fans_Number";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        if(error.equals(result)||result==null){
            return 0;
        }
        return Integer.parseInt(result);
    }
    public static List<FocusData> Get_Follow_Information(int userID){
        methodName="Get_Follow_Information";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        return Json_To_focusData(result);
    }
    public static List<FocusData> Get_Fans_Information(int userID){
        methodName="Get_Fans_Information";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        return Json_To_focusData(result);
    }
    public static OtherPersonData Get_User_All_Info(int userID){
        methodName="Get_User_All_Info";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        return Json_To_OtherPersonData(userID,result);
    }
    public static boolean Add_Focus(int userID,int focusID){
        methodName="Add_Focus";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        soapObject.addProperty("focusID",focusID);
        String result=Get_Post(soapObject);
        return success.equals(result);
    }
    public static boolean Cancel_Focus(int userID,int focusID){
        methodName="Cancel_Focus";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        soapObject.addProperty("focusID",focusID);
        String result=Get_Post(soapObject);
        return success.equals(result);
    }
    public static boolean Check_Follow_Eachohter(int userID,int focusID){
        methodName="Check_Follow_Eachohter";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        soapObject.addProperty("focusID",focusID);
        String result=Get_Post(soapObject);
        return success.equals(result);
    }
    public static List<FocusData> Get_Search_UserInfo(String name){
        methodName="Get_Search_UserInfo";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("name",name);
        String result=Get_Post(soapObject);
        return Json_To_SearchData(result);
    }
    public static boolean is_User_Follow(int userID,int fansID){
        methodName="is_User_Follow";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userName",userID);
        soapObject.addProperty("fansName",fansID);
        String result=Get_Post(soapObject);
        return success.equals(result);
    }
    public static String Get_User_Update_Time(int userID){
        methodName="Get_User_Update_Time";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        return result;
    }
    public static int Get_User_Permission(int userID){
        methodName="Get_User_Permission";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        String result=Get_Post(soapObject);
        return Integer.parseInt(result);
    }

    public static boolean Set_User_Permission(int userID,int permission){
        methodName="Set_User_Permission";
        soapAction=namespace+"/"+methodName;
        SoapObject soapObject=new SoapObject(namespace,methodName);
        soapObject.addProperty("userID",userID);
        soapObject.addProperty("permission",permission);
        String result=Get_Post(soapObject);
        return success.equals(result);
    }
}

File: app/src/main/java/com/example/sunkai/heritage/ConnectWebService/HandlePerson.kt
package com.example.sunkai.heritage.ConnectWebService

import android.util.Base64
import com.example.sunkai.heritage.Activity.LoginActivity.LoginActivity
import com.example.sunkai.heritage.Data.*
import com.example.sunkai.heritage.value.MINI_REPLY
import com.google.gson.Gson
import okhttp3.FormBody

/**个人中心相关的类
 * Created by sunkai on 2018/2/22.
 */
object HandlePerson : BaseSetting() {
    fun GetFollowNumber(userID: Int): Int {
        val getUrl = "$URL/GetFollowNumber?userID=$userID"
        val result = PutGet(getUrl)
        return if (result == ERROR) 0 else result.toInt()
    }

    fun GetFansNumber(userID: Int): Int {
        val getUrl = "$URL/GetFansNumber?userID=$userID"
        val result = PutGet(getUrl)
        return if (result == ERROR) 0 else result.toInt()
    }

    fun GetUserPermission(userID: Int): Int {
        val getUrl = "$URL/GetUserPermission?userID=$userID"
        val result = PutGet(getUrl)
        return if (result == ERROR) -1 else result.toInt()
    }

    fun GetUserFocusAndFansViewPermission(userID: Int): Int {
        val getUrl = "$URL/GetUserFocusAndFansViewPermission?userID=$userID"
        val result = PutGet(getUrl)
        return if (result == ERROR) -1 else result.toInt()
    }

    fun SetUserPermission(userID: Int, permission: Int): Boolean {
        val getUrl = "$URL/SetUserPermission?userID=$userID&permission=$permission"
        val result = PutGet(getUrl)
        return SUCCESS == result
    }

    fun SetUserFocusAndFansViewPermission(userID: Int, permission: Int): Boolean {
        val getUrl = "$URL/SetUserFocusAndFansViewPermission?userID=$userID&permission=$permission"
        val result = PutGet(getUrl)
        return SUCCESS == result
    }

    fun GetUserAllInfo(userID: Int,thisUserID:Int= LoginActivity.userID): UserInfo? {
        val getUrl = "$URL/GetUserAllInfo?userID=$userID"
        val result = PutGet(getUrl)
        if (result == ERROR) {
            return null
        }
        try {
            val data= Gson().fromJson(result, UserInfo::class.java)
            if(data.id==thisUserID){
                data.checked=false
            }else{
                data.checked= IsUserFollow(thisUserID,userID)
            }
            return data
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null

    }

    fun GetFollowInformation(userID: Int): List<FollowInformation> {
        val getUrl = "$URL/GetFollowInformation?userID=$userID"
        val result = PutGet(getUrl)
        if (result == ERROR) {
            return arrayListOf()
        }
        try{
            return fromJsonToList(result, Array<FollowInformation>::class.java)
        }catch (e:Exception){
            e.printStackTrace()
        }
        return arrayListOf()
    }

    fun GetFansInformation(userID: Int):List<FollowInformation>{
        val getUrl="$URL/GetFansInformation?userID=$userID"
        val result = PutGet(getUrl)
        if (result == ERROR) {
            return arrayListOf()
        }
        try{
            return fromJsonToList(result, Array<FollowInformation>::class.java)
        }catch (e:Exception){
            e.printStackTrace()
        }
        return arrayListOf()
    }

    //这两个方法目前服务端还没有对应的方法，预计会在未来解决
    fun GetOtherFollowInformation(userID: Int,otherUserID:Int):List<FollowInformation>{
        val datas= GetFollowInformation(otherUserID)
        datas.map {
            //要把对应的id改为登陆用户的id
            it.focusFocusID=userID
            it.checked= IsUserFollow(userID,it.focusFansID) }
        return datas
    }

    fun GetOtherFansInformation(userID: Int,otherUserID: Int):List<FollowInformation>{
        val datas= GetFansInformation(otherUserID)
        datas.forEach {
            //要把对应的id改为登陆用户的id
            it.focusFocusID=userID
            it.checked= IsUserFollow(userID,it.focusFansID) }
        return datas
    }

    fun AddFocus(userID: Int,focusID:Int):Boolean{
        if(userID==focusID)
            return false
        val getUrl="$URL/AddFocus?userID=$userID&focusID=$focusID"
        val result=PutGet(getUrl)
        return SUCCESS==result
    }

    fun CancelFocus(userID: Int,focusID: Int):Boolean{
        val getUrl="$URL/CancelFocus?userID=$userID&focusID=$focusID"
        val result=PutGet(getUrl)
        return SUCCESS==result
    }

    fun IsUserFollow(userID: Int,fansID:Int):Boolean{
        val gerUrl="$URL/IsUserFollow?userID=$userID&fansID=$fansID"
        return SUCCESS==PutGet(gerUrl)
    }

    fun CheckFollowEachother(userID: Int,focusID: Int):Boolean{
        val getUrl="$URL/CheckFollowEachother?userID=$userID&focusID=$focusID"
        val result=PutGet(getUrl)
        return SUCCESS==result
    }

    fun GetSearchUserInfo(searchName:String,id:Int):List<SearchUserInfo>{
        val getUrl="$URL/GetSearchUserInfo?searchName=$searchName"
        val result=PutGet(getUrl)
        if(result== ERROR){
            return arrayListOf()
        }
        try{
            val searchInfo= fromJsonToList(result, Array<SearchUserInfo>::class.java)
            searchInfo.forEach { it.checked= IsUserFollow(id,it.id) }
            return searchInfo
        }catch (e:Exception){
            e.printStackTrace()
        }
        return arrayListOf()
    }

    fun GetUserImageURL(userID: Int):String?{
        val getUrl="$URL/GetUserImage?userID=$userID"
        val result=PutGet(getUrl)
        return if(result== ERROR||result.isEmpty()){
            null
        }else{
            URL+result
        }
    }

    fun UpdateUserImage(userID: Int,imageByteArray: ByteArray):Boolean{
        val imageString= Base64.encodeToString(imageByteArray,Base64.DEFAULT)
        val postUrl="$URL/UpdateUserImage"
        val formBody=FormBody.Builder().add("userID",userID.toString()).add("image",imageString).build()
        val result=PutPost(postUrl,formBody)
        return result== SUCCESS
    }

    fun AddUserCollection(userID: Int,collectionType:String,typeID:Int):Boolean{
        val getUrl="$URL/AddUserCollection?userID=$userID&typeID=$typeID&type=$collectionType"
        return SUCCESS==PutGet(getUrl)
    }

    fun CancelUserCollect(userID: Int,collectionType:String,typeID:Int):Boolean{
        val getUrl="$URL/CancelUserCollect?userID=$userID&typeID=$typeID&type=$collectionType"
        return SUCCESS==PutGet(getUrl)
    }

    fun CheckIsCollection(userID: Int,collectionType:String,typeID:Int):Boolean{
        val getUrl="$URL/CheckIsCollection?userID=$userID&typeID=$typeID&type=$collectionType"
        val result=PutGet(getUrl)
        return SUCCESS==result
    }

    private fun GetUserCollection(userID: Int,collectionType: String):String{
        val getUrl="$URL/GetUserCollection?userID=$userID&type=$collectionType"
        val result=PutGet(getUrl)
        return result
    }

    fun GetMainCollection(userID: Int,collectionType: String):List<FolkNewsLite>{
        val result=GetUserCollection(userID,collectionType)
        if(ERROR==result)
            return arrayListOf()
        try{
            return fromJsonToList(result, Array<FolkNewsLite>::class.java)
        }catch (e:Exception){
            e.printStackTrace()
        }
        return arrayListOf()
    }

    fun GetFocusOnHeritageCollection(userID: Int,collectionType: String):List<BottomFolkNewsLite>{
        val result=GetUserCollection(userID,collectionType)
        if(ERROR==result)
            return arrayListOf()
        try{
            return fromJsonToList(result, Array<BottomFolkNewsLite>::class.java)
        }catch (e:Exception){
            e.printStackTrace()
        }
        return arrayListOf()
    }

    fun GetFolkColelction(userID: Int,collectionType: String):List<ClassifyDivideData>{
        val result=GetUserCollection(userID,collectionType)
        if(ERROR==result)
            return arrayListOf()
        try{
            return fromJsonToList(result, Array<ClassifyDivideData>::class.java)
        }catch (e:Exception){
            e.printStackTrace()
        }
        return arrayListOf()
    }

    fun GetFindCollection(userID: Int,collectionType: String):List<UserCommentData>{
        val result=GetUserCollection(userID,collectionType)
        if(ERROR==result)
            return arrayListOf()
        try{
            val findResult= fromJsonToList(result, Array<UserCommentData>::class.java)
            findResult.forEach { it.miniReplys=HandleFind.GetUserCommentReply(it.id, MINI_REPLY) }
            return findResult
        }catch (e:Exception){
            e.printStackTrace()
        }
        return arrayListOf()
    }
}